test/integration/signatures/stan_math_sigs.expected

```diff
@@ -14142,7 +14143,60 @@ int[] poisson_rng(row_vector)
 int[] poisson_rng(int[])
 int[] poisson_rng(real[])
 real positive_infinity()
+real pow(int, int)
+real pow(int, real)
+real pow(real, int)
 real pow(real, real)
+vector pow(int, vector)
+vector pow(real, vector)
+vector pow(vector, int)
+vector pow(vector, real)
+vector pow(vector, vector)
+row_vector pow(int, row_vector)
+row_vector pow(real, row_vector)
+row_vector pow(row_vector, int)
+row_vector pow(row_vector, real)
+row_vector pow(row_vector, row_vector)
+matrix pow(int, matrix)
+matrix pow(real, matrix)
+matrix pow(matrix, int)
+matrix pow(matrix, real)
+matrix pow(matrix, matrix)
+real[] pow(int, int[])
+real[] pow(int, real[])
+real[] pow(real, int[])
+real[] pow(real, real[])
+real[] pow(int[], int)
+real[] pow(int[], real)
+real[] pow(int[], int[])
+real[] pow(real[], int)
+real[] pow(real[], real)
+real[] pow(real[], real[])
+vector[] pow(int, vector[])
+vector[] pow(real, vector[])
+vector[] pow(vector[], int)
+vector[] pow(vector[], real)
+vector[] pow(vector[], vector[])
+row_vector[] pow(int, row_vector[])
+row_vector[] pow(real, row_vector[])
+row_vector[] pow(row_vector[], int)
+row_vector[] pow(row_vector[], real)
+row_vector[] pow(row_vector[], row_vector[])
+matrix[] pow(int, matrix[])
+matrix[] pow(real, matrix[])
+matrix[] pow(matrix[], int)
+matrix[] pow(matrix[], real)
+matrix[] pow(matrix[], matrix[])
+real[,] pow(int, int[,])
+real[,] pow(int, real[,])
+real[,] pow(real, int[,])
+real[,] pow(real, real[,])
+real[,] pow(int[,], int)
+real[,] pow(int[,], real)
+real[,] pow(int[,], int[,])
+real[,] pow(real[,], int)
+real[,] pow(real[,], real)
+real[,] pow(real[,], real[,])
```

Comment:
Do we not want

```
vector[,] pow(int, vector[,])
vector[,] pow(real, vector[,])
vector[,] pow(vector[,], int)
vector[,] pow(vector[,], real)
vector[,] pow(vector[,], vector[,])
vector[,,] pow(int, vector[,,])
vector[,,] pow(real, vector[,,])
vector[,,] pow(vector[,,], int)
vector[,,] pow(vector[,,], real)
vector[,,] pow(vector[,,], vector[,,])
```

Reply:
We can nest as deeply as we like, I just wasn't sure what depth we'd want to expose. Should I go up to ```vector[,,]```?

Reply:
With unary functions that support arbitrary nesting with array we tend to go to 7 dimensions.

Reply:
I'm happy to keep the binary vectorisation consistent with the unary vectorisation, I'll add in those signatures and tests

Reply:
Quick question, how deep should the tests go? It looks like the ```exp``` tests only test up to [three levels of nesting](https://github.com/stan-dev/stanc3/blob/master/test/integration/good/function-signatures/math/matrix/exp.stan)

Reply:
Three is fine probably. Things won't change much after that.